### ISSUE TITLE
SqlSyncChangeTrackingProvider  -  configuration error ?

### ISSUE THREAD

**Issue by luc-cap** [Mimetis/Dotmim.Sync#673]
Hi,
     I am usin Dotmim.Sync  in production to sync 4 WEB sites.
and all was fine.  I never tryed: SqlSyncChangeTrackingProvider .

A month ago, I tryed  the  SqlSyncChangeTrackingProvider  solution,
but  probably i LOST some steps.

I  run this command Only  on the Server DB.

ALTER DATABASE AdventureWorks
SET CHANGE_TRACKING = ON
(CHANGE_RETENTION = 14 DAYS, AUTO_CLEANUP = ON)

I call the orchestartor from a WINDOWS Workerservice.

DB server:  SQL Esperes 2019
against a 
DB Client: SQL server 2014

Sync works WELL, but at   day 15  (  change retention is 14 day),
it seems that  all data are  truncated from their Tables.

both databases are empty.

In the Workerservice, if i got an exception --> I I launch immediatly  a sync command with the option:  Reinitialize.

Where am I doing Wrong?

Version ol  Dotmim.Sync  is  0.8.1

 best regards

Luca

**Comment by Mimetis** [comment]
> Sync works WELL, but at day 15 ( change retention is 14 day),
> it seems that all data are truncated from their Tables.

You lost all the datas from your clients databases  ?
Did you lost the server database data also ?

**Comment by luc-cap** [comment]
Unfortunately, I losted the server database data also. 

Should I launch SET CHANGE_TRACKING on clients also ?

**Comment by Mimetis** [comment]
> Should I launch SET CHANGE_TRACKING on clients also ? 

If you using `SqlSyncChangeTrackingProvider` on clients, yes

But first of all, you need to understand how you lost everything on server side

**Comment by luc-cap** [comment]
I have 2 controllers, and 1 Client:
      - sync
      - delete_provisioning

In the sync controller I suppose I made a mistake in  SqlSyncChangeTrackingProvider inizilization (server side)...
instead of passing: the conection and Tables,
I passed: the connection, and a setup object, which declared a list of tables.

Also, in the delete_provisioning controller I de-provisioned,
using SqlSyncProvider instead of SqlSyncChangeTrackingProvider.

Client Side, I was using SqlSyncProvider.

Calling the de-provisioning was probably deadly in the sync phase, and when the server's RETENTION expires it triggers something that clears the data.... Probably first in the Client Db... In fact, if the DB Server were reset first, then in the client DB there would be some data .

At the moment, I prefer to go back and  using SqlSyncProvider.

NJOTE:   Chanching  Provider, tracking Tables  were not  removed...

I had to find It  with this query...

SELECT trac.*, tab.[name]
FROM sys.change_tracking_tables  trac (NOLOCK)
left outer join  sys.tables tab (NOLOCK) ON
trac.object_id = tab.object_id

and disable  CHANGE_TRACKING  for each table

ALTER TABLE [dbo].Store DISABLE CHANGE_TRACKING;  
ALTER TABLE [dbo].StoreMapping DISABLE CHANGE_TRACKING;  
...

Best regards

Luca
